engine/pkg/externalresource/internal/s3/file_path_util.go

```diff
@@ -26,12 +26,17 @@ func getPathPredAlwaysTrue() pathPredFunc {
 	}
 }
 
-func getPathPredByName(target string) pathPredFunc {
+func getPathPredByName(target string, keepPlaceholder bool) pathPredFunc {
```

Comment:
Nit: The meaning of `getPathPredByName` is not clear when it leaves the scope of `removeFilesIf`.

Reply:
Currently `getPathPredByName` is only used in the `removeFilesIf ` function, perhaps we could rename it to a more appropriate name when we extend the application. In addition, do you have a better name to recommend?